### PR TITLE
Igemm codegen v4r1 1x1

### PR DESCRIPTION
**Pull request by shaojiewang**

### CODE REVIEW

File: igemm/igemm_algo_v4r1.py [hunk 1]
```diff
@@ -528,9 +534,12 @@ def __call__(self):
...
-                self._emit('.set k_x,                   80')
-                self._emit('.set k_end,                 84')
+                if 0: #self.is_1x1:
+                    self._emit('.set k_end,             76')
+                else:
+                    self._emit('.set k_y,                   76')
+                    self._emit('.set k_x,                   80')
+                    self._emit('.set k_end,                 84')
                 self._emit_empty_line()
             return self._get_deferred(), igemm_next_mul(84, 8)   # TODO: karg alignment
```
Comment (carlushuang) [inline igemm/igemm_algo_v4r1.py:544]: 1x1 case is not 84 byte

File: config/igemm_v4r1_dynamic.config [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [codegen]
-arch = 'gfx906'
+arch = 'gfx900'
```
Comment (carlushuang) [inline config/igemm_v4r1_dynamic.config:2]: pls keep to gfx906
and this config file should contains both 1x1 and non 1x1. this file seems only have 1x1 now, and only left with 1 config.

File: igemm/igemm_algo_v4r1.py [hunk 3]
```diff
@@ -1666,21 +1635,19 @@ def emit_v4r1_dynamic_kernel(mc, tunable_dicts):
...
-        #     conv_param = conv_param_t(64, 1, 64, 56, 56, 256, 1, 1,  0,  0,  1,  1,  1,  1,  0,  0, CONV_DIRECTION_FWD)
-        #     conv_param.dump()
-        #     for tid in range(64):
-        #         dynamic_index = v4r1_dynamic_get_dynamic_index(igemm_tunable_parameter_t(tunable_dict), conv_param, tid, 0)
-        #         print('tid:{}, v_in_os:{}, v_wei_os:{}, v_out_os:{}, v_sst_a_os:{}, v_sst_b_os:{}, v_sld_a_os:{}, v_sld_b_os:{}'.format(
-        #             tid, dynamic_index.v_in_os, dynamic_index.v_wei_os, dynamic_index.v_out_os, dynamic_index.v_sst_a_os, dynamic_index.v_sst_b_os,
-        #             dynamic_index.v_sld_a_os, dynamic_index.v_sld_b_os
-        #         ))
\ No newline at end of file
+        if 1:
```
Comment (carlushuang) [inline igemm/igemm_algo_v4r1.py:1638]: ideally no "if 1" is needed

File: igemm/igemm_base.py [hunk 4]
```diff
@@ -205,9 +205,17 @@ def igemm_encode_v4r1_kernel_name(tunable):
     thread_tile_m                     = gemm_m_repeat * gemm_m_per_thread_subc
     thread_tile_n                     = gemm_n_repeat * gemm_n_per_thread_subc
 
-    return 'igemm_v4r1_dynamic_' + '{}x{}x{}_{}x{}_{}x{}x{}x{}x{}x{}_{}x{}x{}x{}_{}x{}'.format(
-                k_per_block, b_per_block*gemm_n_repeat*gemm_n_per_thread_subc, e_per_block, 
-                thread_tile_m, thread_tile_n,
-                gemm_m_per_thread_subc,gemm_m_level0_cluster,gemm_m_level1_cluster,gemm_n_per_thread_subc,gemm_n_level0_cluster,gemm_n_level1_cluster,
-                in_block_copy_cluster_lengths_e,in_block_copy_cluster_lengths_n1,in_block_copy_cluster_lengths_b,in_block_copy_cluster_lengths_n2,
-                wei_block_copy_cluster_lengths_e,wei_block_copy_cluster_lengths_k)
+    if is_1x1:
```
Comment (carlushuang) [inline igemm/igemm_base.py:208]: prefix = 'igemm_v4r1_1x1_dynamic' if 1x1 else 'igemm_v4r1_dynamic', then return '{}'.format(prefix) + '{}x{}x{}_{}x{}_{}x{}x{}x{}x{}x{}_{}x{}x{}x{}_{}x{}'.....

File: config/igemm_v4r1_dynamic.config (hunk 2 above)
Comment (shaojiewang) [reply config/igemm_v4r1_dynamic.config]: Ok, modified by the last commit

File: igemm/igemm_algo_v4r1.py (hunk 3 above)
Comment (shaojiewang) [reply igemm/igemm_algo_v4r1.py]: ok, removed by the last commit.

File: igemm/igemm_base.py (hunk 4 above)
Comment (shaojiewang) [reply igemm/igemm_base.py:208]: ok, modified by the last commit

File: igemm/igemm_algo_v4r1.py (hunk 1 above)
Comment (shaojiewang) [reply igemm/igemm_algo_v4r1.py]: use 76 for 1x1 case instead

**Comment by shaojiewang** [comment]
after talked with Carlus, TODO-list:
1.use git merge to update branch code.
2.push config file.
3.add 1x1 special config in .config file to control.
4.generate 1x1 kernel in igemm_codegen.py->igemm_v4r1_emit() level.

**Comment by shaojiewang** [comment]
> after talked with Carlus, TODO-list:
> 1.use git merge to update branch code.
> 2.push config file.
> 3.add 1x1 special config in .config file to control.
> 4.generate 1x1 kernel in igemm_codegen.py->igemm_v4r1_emit() level.

Done by latest commit.

**Pull request merged**